hail/src/main/scala/is/hail/expr/ir/TableWriter.scala

```diff
@@ -654,7 +671,7 @@ case class TableNativeFanoutWriter(
           keyFields,
           s"$targetPath/rows/parts/",
           Some(s"$targetPath/index/" -> keyPType),
-          if (stageLocally) Some(ctx.localTmpdir) else None
+          if (stageLocally) Some(Path.of(ctx.localTmpdir, s"hail_staging_tmp_${UUID.randomUUID()}", "rows", "parts")) else None
```

Comment:
I've noticed we defer to `String`s for files and paths and so on. I have been bitten by this too, hence prefer to use stronger types to encode at the type level that this is indeed a path, not a generic string. Let me know your thoughts, realise this goes against the grain somewhat but if you like it perhaps we can adopt this stuff more.

Reply:
Honestly, I prefer this. We need to take some amount of care for the semantics of cloud storage object names vs filesystem paths, but that is again a way that types come in handy. Thanks for this.